src/app-layer-types.h

```diff
@@ -0,0 +1,31 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#ifndef SURICATA_APP_LAYER_TYPES_H
+#define SURICATA_APP_LAYER_TYPES_H
+
+#include <stdint.h>
+
+typedef enum SCAppLayerEventType {
+    APP_LAYER_EVENT_TYPE_TRANSACTION = 1,
```

Comment:
This was in rust, why is it in C now ?

Reply:
To avoid circular reference I think.  With bindgen we should really think about who *owns* structures and define them there.  I think we defined too many C structures in Rust so we could not repeat ourselves and generate them back out to C. But its a bit of an anti-pattern I suppose.  With bindgen we can now define structures where they belong.

Reply:
Might have also been an artifact of the first version where we auto-generated the bindings on the fly, so we have a chicken and egg problem (circular reference). By generating the bindings and checking them, some of these problems get masked away.